### DISCUSSION TITLE
Can we have empty OS images?

### DISCUSSION THREAD

**Discussion by zooba** [actions/virtual-environments#9388] [General questions]
(Couldn't see a template to create a feature request in the issue tracker, but feel free to move there if you have the power to do so.)

Some of my builds test the installers for software that is included in the default images. Because it is already installed on those images, it's not possible for me to run the tests I need. It would be great to have optional images that do not have all the included software, but instead rely on installing artifacts from previous build stages.

Using self-hosted runners isn't a great option, as if the uninstall fails, the persistent runner will fail repeatedly. Having fresh machines each time is a really important step in this kind of task.

I mainly need this in Azure Pipelines, but I don't see why it wouldn't be useful in GHA too.

**Comment by mikhailkoliada** [comment]
Hello! There were multiple asks of providing slim images (images with the reduced number of packages installed), and why we understand the need we have no plans of providing those, currently.

**Reply by zooba** [reply]
Out of interest, what would you need to change those plans?

**Comment by nulano** [comment]
> Some of my builds test the installers for software that is included in the default images.

While it is not ideal, you may be able to test it in a docker container based on the `mcr.microsoft.com/windows/servercore:ltsc2022` docker image, which should be available in `windows-2022`: https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#cached-docker-images.

For example, in Pillow we need to test that the built Python wheels do not have a dependency on Visual C++ Redistributable being installed and use a Docker container like so:
https://github.com/python-pillow/Pillow/blob/274924e64f8b53f46d04b122fe5d959f848a99b0/.github/workflows/wheels.yml#L207-L213

**Reply by zooba** [reply]
Thanks, great approach! Unfortunately, it seems that MSI (Windows Installer) doesn't work properly inside of these containers, and [neither does PowerShell](https://github.com/PowerShell/PowerShell/issues/5536) or `reg.exe`, so it's not going to work well for us. It seems we need a proper VM for our installer.
